e2e/testdata.go

```diff
@@ -0,0 +1,143 @@
+	if err != nil {
+		return err
+	}
+	return exec.Stream(remotecommand.StreamOptions{
+		Stdin:  stdin,
+		Stdout: stdout,
+		Stderr: stderr,
+		Tty:    tty,
+	})
+}
+
+// ExecWithOptions executes a command in the specified container,
+// returning stdout, stderr and error. `options` allowed for
+// additional parameters to be passed.
+func ExecWithOptions(options ExecOptions) (string, string, error) {
+	if !options.Quiet {
+		fmt.Printf("ExecWithOptions %+v", options)
+	}
+	const tty = false
+
+	fmt.Printf("ExecWithOptions: Client creation")
+	req := clientset.CoreV1().RESTClient().Post().
+		Resource("pods").
+		Name(options.PodName).
+		Namespace(options.Namespace).
+		SubResource("exec").
+		Param("container", options.ContainerName)
+
+	req.VersionedParams(&k8sv1.PodExecOptions{
+		Container: options.ContainerName,
+		Command:   options.Command,
+		Stdin:     options.Stdin != nil,
+		Stdout:    options.CaptureStdout,
+		Stderr:    options.CaptureStderr,
+		TTY:       tty,
+	}, GetParameterCodec())
+
+	var stdout, stderr bytes.Buffer
+	fmt.Printf("ExecWithOptions: execute(POST %s)", req.URL())
+	err := execute("POST", req.URL(), options.Stdin, &stdout, &stderr, tty)
+	if options.PreserveWhitespace {
+		return stdout.String(), stderr.String(), err
+	}
+	return strings.TrimSpace(stdout.String()), strings.TrimSpace(stderr.String()), err
+}
+
+func execCommandInContainerWithFullOutput(podName, containerName string, command []string) (string, string, error) {
+	return ExecWithOptions(ExecOptions{
+		Command:            command,
+		Namespace:          testNamespace,
+		PodName:            podName,
+		ContainerName:      containerName,
+		Stdin:              nil,
+		CaptureStdout:      true,
+		CaptureStderr:      true,
+		PreserveWhitespace: false,
+	})
+}
+
+// execCommandInContainer executes the command in container.
+func execCommandInContainer(podName, containerName string, command string) error {
+	cmd := []string{
+		"sh",
+		"-c",
+		command,
+	}
+	_, stderr, err := execCommandInContainerWithFullOutput(podName, containerName, cmd)
+	fmt.Printf("Exec stderr: %q", stderr)
+	if err != nil {
+		return fmt.Errorf("failed to execute command in pod %v, container %v: %v",
+			podName, containerName, err)
+	}
+	return nil
+}
+
+// execCommandInPod executes the command in pod.
+func execCommandInPod(pod *k8sv1.Pod, cmd string) error {
+	gomega.Expect(pod.Spec.Containers).NotTo(gomega.BeEmpty())
+	return execCommandInContainer(pod.Name, pod.Spec.Containers[0].Name, cmd)
+}
+
+// writeDataInPod writes the data to pod.
+func writeDataInPod(pod *k8sv1.Pod, mode string) error {
+	gomega.Expect(pod.Spec.Containers).NotTo(gomega.BeEmpty())
+	var filePath string
+	if mode == "file" {
+		filePath = pod.Spec.Containers[0].VolumeMounts[0].MountPath + "/test"
+	} else {
+		filePath = pod.Spec.Containers[0].VolumeDevices[0].DevicePath
+	}
+
+	command := fmt.Sprintf("dd if=/dev/zero of=%s bs=1M count=10 status=none", filePath)
```

Comment:
Please write something else to the file and check the contents in validateDataInPod.